File: react-native-meridian-maps/ios/MeridianMapViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTBridge.h>
#import <React/RCTUIManager.h>

// Bridge Swift view manager to React Native and expose props/commands
@interface RCT_EXTERN_MODULE(MeridianMapViewManager, RCTViewManager)

// Props set directly on the underlying view (MeridianMapContainer)
RCT_EXPORT_VIEW_PROPERTY(appId, NSString)
RCT_EXPORT_VIEW_PROPERTY(mapId, NSString)
RCT_EXPORT_VIEW_PROPERTY(appToken, NSString)

// Commands (first arg must be reactTag)
RCT_EXTERN_METHOD(startRoute:(nonnull NSNumber *)reactTag
                  placemarkID:(NSString *)placemarkID)

@end

File: react-native-meridian-maps/ios/MeridianMapViewManager.swift
import Foundation
import React

@objc(MeridianMapViewManager)
final class MeridianMapViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool { true }
  override func view() -> UIView! { MeridianMapContainer() }

  @objc func setAppId(_ reactTag: NSNumber, value: NSString) {
    bridge.uiManager.addUIBlock { _, registry in
      (registry?[reactTag] as? MeridianMapContainer)?.appId = value as String
    }
  }

  @objc func setMapId(_ reactTag: NSNumber, value: NSString) {
    bridge.uiManager.addUIBlock { _, registry in
      (registry?[reactTag] as? MeridianMapContainer)?.mapId = value as String
    }
  }

  @objc func setAppToken(_ reactTag: NSNumber, value: NSString) {
    bridge.uiManager.addUIBlock { _, registry in
      (registry?[reactTag] as? MeridianMapContainer)?.appToken = value as String
    }
  }
}
